node/src/main/scala/coop/rchain/node/runtime/NetworkServers.scala

```diff
@@ -0,0 +1,212 @@
+import coop.rchain.node.diagnostics.{
+  BatchInfluxDBReporter,
+  NewPrometheusReporter,
+  UdpInfluxDBReporter
+}
+import coop.rchain.node.model.repl.ReplGrpcMonix
+import coop.rchain.node.web.ReportingRoutes.ReportingHttpRoutes
+import coop.rchain.node.{api, web}
+import coop.rchain.sdk.syntax.all._
+import coop.rchain.shared.Log
+import coop.rchain.shared.syntax._
+import fs2.concurrent.Queue
+import io.grpc.Server
+import kamon.Kamon
+import kamon.system.SystemMetrics
+import kamon.zipkin.ZipkinReporter
+import monix.eval.Task
+import monix.execution.Scheduler
+import org.http4s.server
+
+object NetworkServers {
+
+  /**
+    * Creates API network service (gRPC and HTTP). Exposed as a [[Resource]] to gracefully shutdown.
+    */
+  // format: off
+  def create[F[_]
+    /* Execution */   : Monixable: ConcurrentEffect: Timer
+    /* Comm */        : TransportLayer: NodeDiscovery: KademliaStore: RPConfAsk: ConnectionsCell
+    /* Diagnostics */ : Log: Metrics] // format: on
+  (
+      routingMessageQueue: Queue[F, RoutingMessage],
+      grpcServices: GrpcServices,
+      webApi: WebApi[F],
+      adminWebApi: AdminWebApi[F],
+      reportingRoutes: ReportingHttpRoutes[F],
+      nodeConf: NodeConf,
+      kamonConf: Config,
+      grpcScheduler: Scheduler
+  )(implicit scheduler: Scheduler): Resource[F, Unit] = {
+    val GrpcServices(deploySrv, proposeSrv, replSrv) = grpcServices
+    val host                                         = nodeConf.apiServer.host
+    for {
+      nodeAddress <- Resource.eval(RPConfAsk[F].ask.map(_.local.toAddress))
+
+      intServer <- internalServer(nodeConf, replSrv, deploySrv, proposeSrv, grpcScheduler)
+      _         <- Resource.eval(Log[F].info(s"Internal API server started at $host:${intServer.getPort}."))
+
+      extServer    <- externalServer(nodeConf, deploySrv, grpcScheduler)
+      extServerMsg = s"External API server started at $host:${extServer.getPort}."
+      _            <- Resource.eval(Log[F].info(extServerMsg))
+
+      _ <- protocolServer(nodeConf, routingMessageQueue)
+      _ <- Resource.eval(Log[F].info(s"Listening for traffic on $nodeAddress."))
+
+      discovery <- discoveryServer(nodeConf, grpcScheduler)
+      _         <- Resource.eval(Log[F].info(s"Kademlia RPC server started at $host:${discovery.getPort}."))
+
+      prometheusRep = new NewPrometheusReporter()
+
+      _         <- Resource.eval(webApiServer(nodeConf, webApi, reportingRoutes, prometheusRep)).flatten
+      webApiMsg = s"HTTP API server started at $host:${nodeConf.apiServer.portHttp}."
+      _         <- Resource.eval(Log[F].info(webApiMsg))
+
+      _              <- Resource.eval(adminWebApiServer(nodeConf, webApi, adminWebApi, reportingRoutes)).flatten
+      adminWebApiMsg = s"Admin HTTP API server started at $host:${nodeConf.apiServer.portAdminHttp}."
+      _              <- Resource.eval(Log[F].info(adminWebApiMsg))
+
+      _ <- metricsInit(nodeConf, kamonConf, prometheusRep)
+    } yield ()
+  }
```

Comment:
This is the main point of this PR. Here is now complete initialization of all network servers wrapped in a Resource which when `use`d will be started/opened and at the end (in case of an error) released/closed.

Cancellation is still not completely covered because of main function not defined with helpers from monix (TaskApp).
https://typelevel.org/cats-effect/docs/2.x/datatypes/ioapp